### PR TITLE
New Swap Flow

### PR DESCRIPTION
**Pull request by hammeiam**
Redesigned Swap Flow as per https://www.figma.com/file/hVhPxjcFcchzB9uFMpT4Fd/Saddle?node-id=933%3A1552

PR is broken into 3 commits and I'd advise reviewing this PR in that order:
1) Translations and package updates (boring)
2) Brand new code (spicy!)
  - The React + CSS for the dropdown
  - The input group that uses the dropdown
  - A hook that lets us close the dropdown when the user clicks outside of it. 
  - Aside, this PR introduces css modules which auto-namespace styles.
3) Using the new code to modify existing code
  - Deletes the old `SwapForm` UI
  - Replaces `SwapForm` with new UI in `SwapPage`
  - Add `valueUSD` to the swap form state plus a helper fn to calculate it.
  - Some styling

protip - try out the keyboard navigation 😎

Screenshots:
Screen Shot 2021-03-08 at 10 16 31 PM
Screen Shot 2021-03-08 at 10 15 19 PM
Screen Shot 2021-03-08 at 10 14 56 PM
Screen Shot 2021-03-08 at 10 14 44 PM
Screen Shot 2021-03-08 at 10 14 34 PM
Screen Shot 2021-03-08 at 10 14 22 PM
Screen Shot 2021-03-08 at 10 14 13 PM
Screen Shot 2021-03-08 at 10 27 10 PM

### CODE REVIEW

File: src/hooks/useDetectOutsideClick.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { RefObject, useEffect } from "react"
+// ElementType,
```
Comment (alphastorm) [inline src/hooks/useDetectOutsideClick.ts:2]: remove if unused

**Review by alphastorm — COMMENTED** [review]
🔥 this looks dope man, awesome job!

**Review by alphastorm — APPROVED** [review]

**Comment by sandrasong** [comment]
Looks great!

One small style change. Could you change the font color in dark theme? 
Slice 1

**Review by sandrasong — APPROVED** [review]

**Review by penandlim — APPROVED** [review]
Looks great! Tested everything works on the mainent fork.

**Pull request closed without merging**